lib/cMake.js

```diff
@@ -199,13 +204,46 @@ CMake.prototype.getConfigureCommand = async function (nodeLibDefPath) {
     for (const [key, value] of Object.entries(this.cMakeOptions)) {
         D.push({ [key]: value });
     }
+    if (this.toolset.cppCompilerPath) {
```

Comment:
I am not sure all of these options should be there though, but it is to be discussed.